lib/fileset/default.nix

```
@@ -510,4 +516,107 @@ If a directory does not recursively contain any file, it is omitted from the sto
       # We could also return the original fileset argument here,
       # but that would then duplicate work for consumers of the fileset, because then they have to coerce it again
       actualFileset;
+
+  /*
+    Like [`gitTrackedWith`](#function-library-lib.fileset.gitTrackedWith) with `{}` as the first argument, therefore using the defaults.
+
+    Type:
+      gitTracked :: Path -> FileSet
+
+    Example:
+      # Include all files tracked by the Git repository in the current directory
+      gitTracked ./.
+
+      # Include only files tracked by the Git repository in the parent directory
+      # that are also in the current directory
+      intersection ./. (gitTracked ../.)
+  */
+  gitTracked =
+    /*
+      The [path](https://nixos.org/manual/nix/stable/language/values#type-path) to a local working directory of the Git repository whose tracked files to include.
```

```suggestion
      The [path](https://nixos.org/manual/nix/stable/language/values#type-path) to a local Git repository.
```

- Remove "working directory" because that's less unambiguously the root of the repo.
- Don't state too much of the obvious.

Not sure where to document the return value, but I'll make another suggestion.

---

I do think for correctness we should use working directory, because there's also bare repositories without a working directory, see https://git-scm.com/docs/gitrepository-layout#_description.

I do agree to remove the "whose tracked files to include" though.

---

```suggestion
      The [path](https://nixos.org/manual/nix/stable/language/values#type-path) to a local working directory of the Git repository.
      This directory must contain a `.git` file or subdirectory.
```

---

```suggestion
      The [path](https://nixos.org/manual/nix/stable/language/values#type-path) to the working directory of a local Git repository.
      This directory must contain a `.git` file or subdirectory.
```

- Switch "the" and "a" around, so we don't suggest there's any choice of directories within a repo. (Technically it may have worktrees but that's not even of secondary importance, and can be fully explained by a small vagueness on our end when it comes to the definition of repository.)

If that's not ok, consider

```suggestion
      The [path](https://nixos.org/manual/nix/stable/language/values#type-path) to a local non-bare Git repository.
      This directory must contain a `.git` file or subdirectory.
```

or if you don't mind an association with the worktree feature, we can be really brief:

```suggestion
      The [path](https://nixos.org/manual/nix/stable/language/values#type-path) to a local Git worktree with a valid `.git` file or directory.
```

---

I like the first suggestion, will apply.